src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-form/rgw-user-form.component.ts

```diff
@@ -274,25 +280,31 @@ export class RgwUserFormComponent implements OnInit {
 
   /**
    * Validate the username.
+   * @param {number|Date} dueTime The delay time to wait before the
+   *   API call is executed. This is useful to prevent API calls on
+   *   every keystroke. Defaults to 500.
    */
-  userIdValidator(): AsyncValidatorFn {
+  userIdValidator(dueTime = 500): AsyncValidatorFn {
```

Comment:
This doesn't solves the problem in my opinion as it's only delaying the problem. It reduces it a bit, but the problem is still there. Why not validate on focus loss and before submitting?

Reply:
This is incorrect, if there are keystrokes during this 500ms, then every delayed request before is canceled. If there are no more keystrokes during this time, the last value is submitted via REST call.

Reply:
> Why not validate on focus loss and before submitting?

I want to have feedback as soon as possible during typing. We have a modern UI that supports this, so why not using it.

Reply:
How about firing after 750ms and focus loss? Than you depend on which is fired first.